torch/csrc/nested_tensor/python_nested_tensor.h

```diff
@@ -0,0 +1,442 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <torch/csrc/Device.h>
+#include <torch/csrc/Dtype.h>
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/autograd/python_variable.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/tensor/python_tensor.h>
+#include <torch/csrc/utils/tensor_new.h>
+
+// TODO:
+// - HANDLE_TH_ERRORS
+// - Python exception handling.
+// - Implement NestedSize to avoid expensive python ops in *_nested_size/stride
+// - map and apply functions that accepted JIT-ed functions to avoid unbind
+// - don't make some functions static?
+// - DEBUG enabled checking of constiuents
+
+namespace torch {
+namespace nested_tensor {
+
+using namespace at;
+using namespace torch::autograd;
+
+struct _ListNestedTensor;
+
+struct _VariableNode {
```

Comment:
how is this different than a Variable?

Reply:
I want to minimize the amount of memory necessary to point to a Variable from _NestedNode. _NestedNode may have children, or if it it's a leaf it carries a payload such as Variable, but I don't want each node in the tree to have an empty torch Variable. I might be getting this wrong, but I'd prefer to make these decisions based on profiling later on once everything is cpython compliant. I'll run some basic benchmarks on top of this, but maybe we can land those kinds of improvements in a separate PR?